inc/saip4ext.h

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 20XX Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saip4ext.h
+ *
+ * @brief   This module defines SAI P4 Extension interface
+ */
+
+#if !defined (__SAIP4EXT_H_)
+#define __SAIP4EXT_H_
+
+#include <saitypes.h>
+
+/**
+ * @brief Attribute Id for P4 ext
+ */
+typedef enum _sai_p4ext_entry_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_P4EXT_ENTRY_ATTR_START,
+
+    /**
+     * @brief SAI P4 EXT table id
+     *
+     * @type sai_s8_list_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY | KEY
+     */
+    SAI_P4EXT_ENTRY_ATTR_TABLE_ID = SAI_P4EXT_ENTRY_ATTR_START,
+
+    /**
+     * @brief SAI P4 EXT Match field
+     *
+     * @type sai_s8_list_t
+     * @flags CREATE_AND_SET
+     * @default empty
+     */
+    SAI_P4EXT_ENTRY_ATTR_MATCH_FIELD_ID,
+
+    /**
+     * @brief SAI P4 EXT Action id
+     *
+     * @type sai_s8_list_t
+     * @flags CREATE_AND_SET
+     * @default empty
+     */
+    SAI_P4EXT_ENTRY_ATTR_ACTION_ID,
+
```

Comment:
How about objects like Hash are referenced by other tables as it doesn't have match fields?